cmd/csv2parquet/main.go

```diff
@@ -0,0 +1,435 @@
+		fieldHandler = intHandler(64)
+	case "uint64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: false}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_64)
+		fieldHandler = uintHandler(64)
+	case "float":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_FLOAT)
+		fieldHandler = floatHandler
+	case "double":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
+		fieldHandler = doubleHandler
+	case "int":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		fieldHandler = intHandler(64)
+	case "json":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.JSON = &parquet.JsonType{}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_JSON)
+		fieldHandler = jsonHandler
+	default:
+		return nil, nil, fmt.Errorf("unsupported type %q", typ)
+	}
+
+	fieldHandler = optionalHandler(fieldHandler) // TODO: if we make repetition type configurable, change this to use correct handler.
+
+	return col, fieldHandler, nil
+}
+
+func parseTypeHints(s string) (map[string]string, error) {
+	typeMap := make(map[string]string)
+
+	if s == "" {
+		return typeMap, nil
+	}
+
+	hintsList := strings.Split(s, ",")
+	for _, hint := range hintsList {
+		hint = strings.TrimSpace(hint)
+
+		hintFields := strings.Split(hint, "=")
+		if len(hintFields) != 2 {
+			return nil, fmt.Errorf("invalid type hint %q", hint)
+		}
+
+		fieldName := strings.TrimSpace(hintFields[0])
+		fieldType := strings.TrimSpace(hintFields[1])
+
+		if !isValidType(fieldType) {
+			return nil, fmt.Errorf("invalid parquet type %q", fieldType)
+		}
+
+		typeMap[fieldName] = fieldType
+	}
+
+	return typeMap, nil
+}
+
+var validTypes = map[string]bool{
+	"boolean":    true,
+	"int8":       true,
+	"uint8":      true,
+	"int16":      true,
+	"uint16":     true,
+	"int32":      true,
+	"uint32":     true,
+	"int64":      true,
+	"uint64":     true,
+	"float":      true,
+	"double":     true,
+	"byte_array": true,
+	"string":     true,
+	"int":        true,
+	"json":       true,
+	// TODO: support more data types
+}
+
+func validTypeList() []string {
+	l := make([]string, 0, len(validTypes))
+	for k := range validTypes {
+		l = append(l, k)
+	}
+	sort.Strings(l)
+	return l
+}
+
+var validCodecs = map[string]parquet.CompressionCodec{
+	"none":   parquet.CompressionCodec_UNCOMPRESSED,
+	"snappy": parquet.CompressionCodec_SNAPPY,
+	"gzip":   parquet.CompressionCodec_GZIP,
+}
```

Comment:
the compression codecs are pluggable, we can export a method to get the list of them in the package, instead of having it here, wdyt?

Reply:
Makes sense. I introduced a new function to export all registered block compressors. The list of available compression codecs presented in `csv2parquet` is then derived from it.